tests/topology.rs

```diff
@@ -39,6 +43,63 @@ fn into_message(event: Event) -> String {
         .to_string_lossy()
 }
 
+#[test]
+fn topology_shutdown_while_active() {
```

Comment:
So turns out this test actually passes even without my changes, so it's not actually testing this change really.  Nevertheless I already had it written and it seemed worthwhile so I left it in to just add to our overall coverage.